src/BuildScriptGenerator/Php/VersionProviders/PhpComposerVersionProvider.cs

```diff
@@ -32,14 +33,14 @@ public PlatformVersionInfo GetVersionInfo()
         {
             if (_versionInfo == null)
             {
+                _versionInfo = _onDiskVersionProvider.GetVersionInfo();
                 if (_options.EnableDynamicInstall)
                 {
-                    return _sdkStorageVersionProvider.GetVersionInfo();
+                    var sdkStorageVersionProviderResult = _sdkStorageVersionProvider.GetVersionInfo();
+                    _versionInfo.SupportedVersions.Union(sdkStorageVersionProviderResult.SupportedVersions);
```

Comment:
we don't need union it because _sdkStorageVersionProviderResult_ will cover all versions, they're overlapped.

Reply:
yes .. i was testing something ... and this debug code needs to be removed ... thanks for catching this